integration-cli/docker_cli_rename_test.go

```diff
@@ -38,6 +38,27 @@ func (s *DockerSuite) TestRenameRunningContainer(c *check.C) {
 	c.Assert(name, checker.Equals, "/"+newName, check.Commentf("Failed to rename container %s", name))
 }
 
+func (s *DockerSuite) TestRenameRunningContainerAndReuse(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	out, _ := dockerCmd(c, "run", "--name", "first_name", "-d", "busybox", "top")
+	c.Assert(waitRun("first_name"), check.IsNil)
+
+	newName := "new_name" + stringid.GenerateNonCryptoID()
```

Comment:
Cleanup can happen separately, I just don't think we need this here.
it's something extra that someone has to look at and decide if it's meaningful, and I don't think it is here.